vmdb/spec/support/api_spec_helper.rb

```diff
@@ -116,9 +119,39 @@ def collection_config
     api_config[:collections]
   end
 
-  def action_identifier(type, action)
-    collection_config.fetch_path(type, :resource_actions, :post)
+  def action_identifier(type, action, selection = :resource_actions)
+    collection_config.fetch_path(type, selection, :post)
       .select { |spec| spec[:name] == action.to_s }
       .first[:identifier]
   end
+
+  def collection_action_identifier(type, action)
+    action_identifier(type, action, :collection_actions)
+  end
+
+  def subcollection_action_identifier(type, subtype, action)
+    action_identifier(type, action, "#{subtype}_subcollection_actions".to_sym)
+  end
+
+  def gen_request(action, *hrefs)
+    request = {"action" => action.to_s}
+    request["resources"] = hrefs.collect { |href| {"href" => href} } if hrefs.present?
```

Comment:
This is not a rule, but I always get confused when calling `.present?` on an Array, personally.  I feel like "the Array is there so it's present", even though I know `.present?` just means `!.blank?`.  I usually use `.any?` instead, because to me that's clearer for collections.